### PR TITLE
[DEV-10768] Adds required filters to spending_by_transaction

### PR DESCRIPTION
**Pull request by collinwr**
**Description:**
Previously the `/spending_by_transaction` endpoint only had `keywords` and `award_type_codes` listed as valid filters even though it supported other filters available in Advanced Search. This fixes this inconsistency by adding documentation about the available filters. 

**Requirements for PR merge:**

1. N/A - Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend
4. N/A - Matview impact assessment completed
5. [ ] Frontend impact assessment completed
6. N/A - Data validation completed
7. N/A - Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-10768](https://federal-spending-transparency.atlassian.net/browse/DEV-10768):
    - [ ] Link to this Pull-Request
    - N/A - Performance evaluation of affected (API | Script | Download)
    - N/A - Before / After data comparison

**Area for explaining above N/A when needed:**
```
No code changes other than contracts. 
```

### CODE REVIEW

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md [hunk 1]
```diff
@@ -248,6 +248,93 @@ Returns transaction records which match the keyword and award type code filters.
...
++ `extent_competed_type_codes`: [`A`] (optional, array[string])
++ `treasury_account_components` (optional, array[TreasuryAccountComponentsObject], fixed-type)
++ `object_class` (optional, array[string])
++ `program_activity` (optional, array[number])
++ `def_codes` (optional, array[DEFC], fixed-type)
+    If the `def_codes` provided are in the COVID-19 or IIJA group, the query will only return transactions that meet two requirements:
+    1. The transaction's associated prime award has at least one File C record with one of the supplied DEFCs.
+    2. The matching DEFC's associated public law has an enactment date prior to the transaction's action_date.
+
+### TimePeriodObject (object)
```
Comment (dporth-frb) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md:287]: Does this endpoint support new awards only?

Comment (collinwr) [reply usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md:287]: No, I just dug into this a bit, and it only currently supports `action_date`. I'll update this contract and create a new ticket to add this functionality.

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md [hunk 2]
```diff
@@ -208,51 +208,128 @@ Returns transaction records which match the keyword and award type code filters.
...
+
+### PageMetaDataObject (object)
 + `page`: 1 (required, number)
 + `hasNext`: false (required, boolean)
 + `hasPrevious`: false (required, boolean)
 + `next` (required, number, nullable)
 + `previous` (required, number, nullable)
 
-## TransactionResponse (object)
+### AdvancedFilterObject (object)
```
Comment (ayubshahab) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md:220]: Are we potentially leaving out some filters? I am referring to "legal_entities" and or "recipient_id".

File: dredd-results.txt
[the remark this answers is not in the document]
Comment (collinwr) [reply dredd-results.txt:1]: This PR deletes the file. Good eye seeing it though 😄

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md (hunk 2 above)
Comment (collinwr) [reply usaspending_api/api_contracts/contracts/v2/search/spending_by_transaction.md:220]: I checked the codebase and tried hitting the API. These filters do not seem to be supported by the endpoint.

**Review by aguest-kc — APPROVED** [review]

**Review by ayubshahab — APPROVED** [review]

**Pull request merged**